mutiny-core/src/lib.rs

```diff
@@ -124,10 +141,53 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         Ok(())
     }
 
+    /// Starts a background process that will watch for nostr wallet connect events
+    pub async fn start_nostr_wallet_connect(&self, from_node: PublicKey) {
+        let nostr = self.nostr.clone();
+        let nm = self.node_manager.clone();
+        spawn_local(async move {
```

Comment:
It shouldn't actually be hard to stop here, just throw the atomic bool checker in the loop.